src/components/Card.js

```diff
@@ -1,28 +1,43 @@
 export default class Card {
-  constructor(cardData, cardSelector, handleImageClick) {
+  constructor(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeClick
+  ) {
     this._name = cardData.name;
     this._link = cardData.link;
+    this._id = cardData._id;
+    this.setIsLiked = cardData.isLiked;
     this._cardSelector = cardSelector;
-    this._handleImageClick = handleImageClick;
+    this.handleImageClick = handleImageClick;
+    this.handleDeleteCard = handleDeleteCard;
+    this.handleLikeClick = handleLikeClick;
   }
 
   _setEventListeners() {
     this._cardElement
-      .querySelector(".card__like-button")
+      .querySelector(".card__image")
       .addEventListener("click", () => {
-        this._handleLikeIcon();
+        this.handleImageClick({ name: this._name, link: this._link });
       });
 
     this._cardElement
       .querySelector(".card__delete-button")
       .addEventListener("click", () => {
-        this._handleDeleteCard();
+        this.handleDeleteCard(this);
       });
 
     this._cardElement
-      .querySelector(".card__image")
+      .querySelector(".card__like-button")
+      .addEventListener("click", () => {
+        this.handleLikeClick(this);
+      });
+    this._cardElement
+      .querySelector(".card__like-button")
       .addEventListener("click", () => {
-        this._handleImageClick({ name: this._name, link: this._link });
+        this._handleLikeIcon();
```

Comment:
Please, delete the 2nd like handler. You have it in line 35

Reply:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.


`_handleLikeIcon` should be **public** (without `_`) to be able to call in in a `then` block in `index.js`